pkg/vpcmodel/nodesExplainability.go

```diff
@@ -131,6 +133,20 @@ func (c *VPCConfig) GetNodesWithinAddress(ipAddress string) (networkInterfaceNod
 	return networkInterfaceNodes, nil
 }
 
+// getNodesOfVsi is given a string name or UID of VSI, and
+// returns the list of all nodes with this vsi name or UID
+func (c *VPCConfig) GetNodesOfVsi(vsi string) (vsiNodes []Node, err error) {
+	for _, nodeSet := range c.NodeSets {
+		if nodeSet.Kind() == ResourceTypeVSI && (nodeSet.Name() == vsi || nodeSet.UID() == vsi) {
```

Comment:
Not exactly. What I meant was that there are at least two different `NodeSets` such that in each of them the given name equals the Name or the UID. Its not likely it will be in the same `NodeSets`

Reply:
Seems the comparison is for different NodeSets

Reply:
I do keep the previous nodeSet..

Reply:
So please in the error message notify which are the two NodeSets